src/sign-eip7702-authorization.ts

```diff
@@ -126,4 +130,30 @@ function validateEIP7702Authorization(authorization: EIP7702Authorization) {
   if (isNullish(nonce)) {
     throw new Error('Missing nonce parameter');
   }
+
+  if (
+    typeof chainId !== 'number' ||
+    chainId >= 2 ** CHAIN_ID_MAX_BITLENGTH ||
+    chainId < 0
+  ) {
+    throw new Error(
+      `Invalid chainId: must be a non-negative number less than 2^${CHAIN_ID_MAX_BITLENGTH}`,
+    );
+  }
+
+  if (
+    typeof nonce !== 'number' ||
+    nonce >= 2 ** NONCE_MAX_BITLENGTH ||
+    nonce < 0
+  ) {
+    throw new Error(
+      `Invalid nonce: must be a non-negative number less than 2^${NONCE_MAX_BITLENGTH}`,
+    );
+  }
+
+  if (!isValidHexAddress(contractAddress as Hex)) {
```

Comment:
For the record: This function does verify the address length too.

See: https://github.com/MetaMask/utils/blob/v11.3.0/src/hex.ts#L15-L18 (`{40}` in hex-format, each bytes is 2-digits/chars long, which makes it a length of 20 bytes).